### PR TITLE
SNOW-196640 Added Pipe Creation Telemetry and Removed Previous Telemetry

### PR DESCRIPTION
**Pull request by sfc-gh-zli**
Added a telemetry object
**pipeCreation** is sent at the creation of a pipe and contains:
1. Is the pipe reusing table/stage/pipe
2. Count of files that are on the stage
3. Count of files that are going to be reprocessed

**Two dashboards**:
**pipe status monitor**: https://app.snowflake.com/us-west-2/snowhouse/kafka-connector-pipe-status-dJeFMIbye
We are able to derive:
1. Time spent on ingestion and on Kafka
2. Processed offset, flushed offset, committed offset, and purged offset
3. Count of files that are on the stage, waiting for ingestion, and moved to table stage

**app info board**: https://app.snowflake.com/us-west-2/snowhouse/kafka-connector-app-status-dL5NosxdZ
We are able to derive:
1. Connector version, Kafka version, start time of the connector
2. Number of partitions and to which Sink Task was a partition assigned
3. Any fatal errors with error messages

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryBasicInfo.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.snowflake.kafka.connector.internal;
+
+public class SnowflakeTelemetryBasicInfo {
```
Comment (sfc-gh-bli) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryBasicInfo.java:3]: make it `abstract class` ?

**Review by sfc-gh-bli — APPROVED** [review]

File: src/test/java/com/snowflake/kafka/connector/ConnectorIT.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 import java.util.Map;
 import java.util.Set;
 
-public class ConnectorTest
+public class ConnectorIT
```
Comment (sfc-gh-zli) [inline src/test/java/com/snowflake/kafka/connector/ConnectorIT.java:15]: Move the test to Internal Test (IT) as it requires Snowflake Connection to run.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeErrors.java [hunk 3]
```diff
@@ -414,7 +414,7 @@ public SnowflakeKafkaConnectorException getException(
   {
     if(telemetryService != null)
     {
-      telemetryService.reportKafkaFatalError(getCode());
+      telemetryService.reportKafkaFatalError(getCode() + msg);
```
Comment (sfc-gh-sshankar) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeErrors.java:417]: Is there a limit on the size of msg

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java [hunk 4]
```diff
@@ -343,13 +334,13 @@ private void init(long recordOffset)
     {
       logInfo("init pipe: {}", pipeName);
       // wait for sinkConnector to start
-      createTableAndStage();
+      SnowflakeTelemetryPipeCreation pipeCreation = createTableAndStage();
```
Comment (sfc-gh-sshankar) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java:337]: Where is event reported ?

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeErrors.java (hunk 3 above)
Comment (sfc-gh-zli) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeErrors.java:417]: No. Should I truncate the msg to the first 500 characters? The first several hundreds characters should contains enough information for debugging.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java (hunk 4 above)
Comment (sfc-gh-zli) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java:337]: In line 386 of this file

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java [hunk 5]
```diff
@@ -54,154 +53,69 @@ public void setAppName(final String name)
   }
 
   @Override
-  public void reportKafkaStart(final long startTime, final int maxTasks)
-  {
-    ObjectNode msg = MAPPER.createObjectNode();
```
Comment (sfc-gh-sshankar) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java:59]: Is this code not used any more ?

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java [hunk 6]
```diff
@@ -9,10 +9,12 @@
 import net.snowflake.client.jdbc.telemetry.TelemetryClient;
 import net.snowflake.client.jdbc.telemetry.TelemetryData;
 import net.snowflake.client.jdbc.telemetry.TelemetryUtil;
+import org.apache.kafka.common.utils.AppInfoParser;
 
-import java.io.IOException;
```
Comment (sfc-gh-sshankar) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java:13]: There's a lot of change sin this file. Are these related to pipe creation telemetry ?

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java (hunk 5 above)
Comment (sfc-gh-zli) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java:59]: This is still in use, it is in line 70 in the file. Git diff marked this function as a deletion, which is not accurate. It is the same case with reportKafkaStop.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java (hunk 4 above)
Comment (sfc-gh-sshankar) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java:337]: It's very hard to understand the life-cycle of this record. i.e. where it is created where it is updated and where it is used.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java (hunk 6 above)
Comment (sfc-gh-zli) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java:13]: Most changes in this PR is to remove current telemetry that will be covered by new telemetry.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java (hunk 4 above)
Comment (sfc-gh-sshankar) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java:337]: It would be much more understandable if the record was created and then "used" in the same function/block

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java [hunk 7]
```diff
@@ -229,20 +146,23 @@ private String getAppName()
...
-    KAFKA_NONFATAL_ERROR("kafka_nonfatal_error"),
-    KAFKA_USAGE("kafka_usage"),
-    KAFKA_CREATE_TABLE("kafka_create_table"),
-    KAFKA_REUSE_TABLE("kafka_reuse_table"),
-    KAFKA_CREATE_STAGE("kafka_create_stage"),
-    KAFKA_REUSE_STAGE("kafka_reuse_stage"),
-    KAFKA_CREATE_PIPE("kafka_create_pipe"),
-    KAFKA_FILE_FAILED("kafka_file_failed"),
-    KAFKA_SNOWFLAKE_THROTTLE("kafka_snowflake_throttle");
+    KAFKA_PIPE_USAGE("kafka_pipe_usage"),
```
Comment (sfc-gh-sshankar) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java:164]: Is this being used anywhere

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java (hunk 4 above)
Comment (sfc-gh-zli) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java:337]: pipeCreation collects information in multiple stages (createTableAndStage, recover, startCleaner) of the pipe starting process.  I think I can move uploading pipeCreation telemetry from inside startCleaner to outside this function. So that the life cycle of this object is obvious in this function. We create the object, then update its value in multiple functions then upload it.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java (hunk 7 above)
Comment (sfc-gh-zli) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryServiceV1.java:164]: This is for uploading pipeStatus object, which is removed in this PR. I should have removed this in this PR as well.

**Review by sfc-gh-sshankar — APPROVED** [review]
